tools/src/scie_pants/install_pants.py

```diff
@@ -78,22 +79,30 @@ def install_pants_from_pex(
     ptex: Ptex,
     extra_requirements: Iterable[str],
     find_links: str | None,
+    bootstrap_urls_path: str | None,
 ) -> None:
     """Installs Pants into the venv using the platform-specific pre-built PEX."""
     uname = os.uname()
-    major, minor = sys.version_info[:2]
-    pex_name = (
-        f"pants.{version}-cp{major}{minor}-{uname.sysname.lower()}_{uname.machine.lower()}.pex"
-    )
+    pex_name = f"pants.{version}-cp39-{uname.sysname.lower()}_{uname.machine.lower()}.pex"
+
     pex_url = f"https://github.com/pantsbuild/pants/releases/download/release_{version}/{pex_name}"
+    if bootstrap_urls_path:
+        bootstrap_urls = json.loads(Path(bootstrap_urls_path).read_text())
+        urls_info = bootstrap_urls["ptex"]
+        pex_url = urls_info.get(pex_name)
```

Comment:
What's the behaviour/diagnostics if the JSON file is invalid? Eg invalid syntax, no `ptex` key, `ptex` key maps to non-object (eg `null`), the pex name maps to a non-string

Reply:
Everything but the last one, I assume is already handled by `scie` machinery itself. The very last one I assume is "garbage-in-garbage-out".

Reply:
Hm, I think we can aspire to have better diagnostics in our user-friendly interactive tool than just "garbage-in-garbage-out". I tried the various cases like invalid JSON and so on, and they  just gave the raw `json` exception summary. There's a backtrace in `install.log`, but we can do better, e.g. you've already done great with the diagnostic for the key being missing.   I'd love to see that for other cases too. 😄 

As a relatively low effort improvement, we could do a whole-sale wrapping of any exception, plus checking for a string would be fine as a start:

```python
try:
    text = json.loads(Path(bootstrap_urls_path).read_text())
    urls_info = bootstrap_urls["ptex"]
    pex_url = urls_info.get(pex_name)
except Exception as e:
    raise ValueError(f"Failed to parse PANTS_BOOTSTRAP_URLS file '{bootstrap_urls_path}': it must be JSON like `{{"ptex": {{"{pex_name}": "$url", ...}}}}`: {e}") from e

if pex_url is None:
    ...

if not isinstance(pex_url, str):
    raise ValueError(
        f"Found invalid '{pex_name}' value {pex_url!r} in PANTS_BOOTSTRAP_URLS file '{bootstrap_urls_path}': it must be a URL as a string"
    )
```

One possibility would be pulling in a library like Pydantic or attrs or similar to not have to do all this manual validation/parsing, but that's probably not worth it if it's just this one place.

Reply:
When you tested invalid JSON, did you test it end-to-end? I expected the scie bits to have already errored by the time we get here on things like invalid JSON

Reply:
Either way, I'll bang on this some more

Reply:
Yeah, you're right, not fully-end-to-end, the scie/ptex stuff was already installed, so maybe not a fully realistic scenario.

Reply:
OK I tested, and anything but the value being the wrong type is already handled by the `scie` plumbing through generic errors.